dask/array/ufunc.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import absolute_import, division, print_function
+
+from operator import getitem
+
+import numpy as np
+
+from toolz.curried import merge
+from .core import Array, elemwise
+from .. import core
+from ..utils import skip_doctest
+
+
+def __array_wrap__(numpy_ufunc, x, *args, **kwargs):
+    return x.__array_wrap__(numpy_ufunc(x, *args, **kwargs))
+
+
+def wrap_elemwise(numpy_ufunc, array_wrap=False):
```

Comment:
I don't think this will pickle properly, which will likely be an annoyance later (see https://github.com/pydata/xarray/issues/901).